db/frontend.go

```diff
@@ -539,7 +544,7 @@ func GetHistoricPrices(currency string) (map[uint64]float64, error) {
 	genesisTime := time.Unix(int64(utils.Config.Chain.GenesisTimestamp), 0)
 
 	for _, d := range data {
-		day := uint64(d.Ts.Sub(genesisTime).Hours()) / 24
+		day := uint64(d.Ts.Add(utils.Day).Sub(genesisTime).Hours()) / 24
```

Comment:
Slight performance suggestion, rather than retrieving and mapping every historic price as needed, we should either store the map atomically and update it in a service so we don't always have to fetch it from the db.

Or modify the query so we get the historic price directly from the database instead of querying every historic price.

Reply:
I rewrote the function/query to only look for a specific day.